### PR TITLE
feat: register dataplex as provider

### LINKED ISSUE #319: Register Dataplex policy-tags as provider
**Summary**
Register DataCatalog policy tags as a provider so users can raise requests for policy tags with `roles/datacatalog.categoryFineGrainedReader` permission

### PR DESCRIPTION
**Pull request by singhvikash11**
closes #319

### CODE REVIEW

File: plugins/providers/policytag/client.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	}
+	return &policyTagClient{
+		policyManager: policyManager,
+		projectId:     projectID,
+	}, nil
+}
+
+func (p *policyTagClient) GetPolicies(ctx context.Context) ([]*Policy, error) {
+	taxonomies := p.policyManager.ListTaxonomies(ctx, &pb.ListTaxonomiesRequest{
+		Parent: p.toParent(p.projectId, US),
```
Comment (bsushmith) [inline plugins/providers/policytag/client.go:35]: Any particular reason, why location is hardcoded as `US` ?

File: plugins/providers/policytag/config.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	decryptedCredentials, err := decryptor.Decrypt(c.ServiceAccountKey)
+	if err != nil {
+		return err
+	}
+
+	c.ServiceAccountKey = decryptedCredentials
+	return nil
+}
+
+// Permission is for mapping role into policy tag fin-grained-reader permissions
```
Comment (bsushmith) [inline plugins/providers/policytag/config.go:56]: can you fix the typo in the comment here

File: plugins/providers/policytag/client.go (hunk 1 above)
Comment (singhvikash11) [reply plugins/providers/policytag/client.go]: take resource name as projects/{project_id}/locations/{location}

File: plugins/providers/policytag/config.go (hunk 2 above)
Comment (singhvikash11) [reply plugins/providers/policytag/config.go]: done

File: docs/docs/providers/policy_tag.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+## Access Management
+
+Access can be given at the policy-tag level as those allowed to be managed through these DataCatalog APIs:
+
+## Config
+
+#### YAML Representation
+
+```yaml
+type: policy_tag
```
Comment (mabdh) [inline docs/docs/providers/policy_tag.md:41]: I don't think naming it as `policy_tag` type or putting this inside `policytag` package would be good. 
maybe just calling it `dataplex` ?
We have tag and policy under resources anyway.

File: docs/docs/providers/policy_tag.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
+# Data Catalog Policy tag
```
[the remark this answers is not in the document]
Comment (singhvikash11) [reply docs/docs/providers/policy_tag.md]: done

File: docs/docs/providers/policy_tag.md (hunk 3 above)
Comment (singhvikash11) [reply docs/docs/providers/policy_tag.md]: done

**Comment by mabdh** [comment]
@bsushmith @utsav14nov there is a newly merge changes to enable tracing, should we implement that as well for this provider? or can we add another issue to track on?

**Comment by bsushmith** [comment]
@mabdh @utsav14nov Lets have a separate issue for tracking - `enabling tracing in dataplex provider`

**Review by haveiss — APPROVED** [review]

**Pull request merged**